lib/index.js

```diff
@@ -4,165 +4,88 @@
-    const repoName = payload.repository.full_name;
-
-    if (!hasRelationship(repoName) || !pushPayloadHasCopyableCommits(payload)) {
-      return;
-    }
-
-    const PR = await getPRFromPushPayload(payload);
+export default function(app) {
+  getProbotRepoNames(app)
+    .then(init)
+    .catch(console.error);
 
-    if (PR) {
-      const partnerPR = await getPartnerPR(PR);
+  app.on('*', async context => {
+    // for reuse, saves the graphql client authenticated for this installation
+    saveAppGithubClient(context);
+  });
 
-      if (partnerPR) {
-        await queue(genQueueKey(partnerPR, 'push'), () => {
-          if (hasParent(repoName)) {
-            syncParentPR(payload, PR, partnerPR);
-          } else {
-            syncChildPR(payload, PR, partnerPR);
-          }
-        });
+  app.on(
+    [
+      'pull_request.closed',
+      'pull_request.edited',
+      'pull_request.opened',
+      'pull_request.reopened',
+      'pull_request.synchronize',
+    ],
+    async ({payload}) => {
+      if (payload.sender.id === (await getGithubId())) return;
+
+      const pullRequest = {
+        number: payload.pull_request.number,
+        repoName: payload.repository.full_name,
+      };
+
+      switch (payload.action) {
+        case 'closed':
+        case 'reopened':
+          return syncPR(pullRequest, ['state']);
+        case 'edited':
+          return syncPR(pullRequest, ['meta']);
+        case 'opened':
+        case 'synchronize':
+          return syncPR(pullRequest, ['commits', 'state']);
       }
-    }
-  }
-
-  // child changes authored from parent repo -> child repo
-  // child PR merges -> merge parent PR
-  // parent PR merges -> merge child PR
-  async function onPullRequestMerge({payload}) {
-    const PR = payload.pull_request;
-    const repoName = payload.repository.full_name;
-
-    if (!PR.merged || !hasRelationship(repoName)) {
-      return;
-    }
-
-    const partnerPR = await getPartnerPR(PR);
-
-    if (partnerPR) {
-      await queue(genQueueKey(partnerPR, 'merge'), () => mergePR(partnerPR));
-    } else if (hasChildren(repoName)) {
-      await syncChildRepos(PR);
-    }
-  }
-
-  // grouped parent PR status -> child PR
-  // grouped child PR status -> parent PR
-  async function onCommitStatusChange({payload}) {
-    const repoName = payload.repository.full_name;
-
-    if (!hasRelationship(repoName)) {
-      return;
-    }
-
-    const PR = await getPRFromStatusPayload(payload);
-
-    if (PR) {
-      const partnerPR = await getPartnerPR(PR);
+    },
+  );
 
-      if (partnerPR) {
-        const statusContext =
-          STATUS_CONTEXTS[hasParent(repoName) ? 'child' : 'parent'];
+  app.on('status', async ({payload}) => {
+    if (isSyncStatusContext(payload.context)) return;
+
+    // throttle this since it's sometimes triggered more than 20 times within
+    // a window of ~10 seconds
+    const handler = throttleWebhook(payload.sha, 10000, async payload => {
```

Comment:
I didn't see a callsite for this function. Should we remove it if it's not used?

Reply:
[hard to see it buried in the diff](https://github.com/uber-workflow/probot-app-monorepo-sync/pull/10/files/a74429cf3fad567741daebfef3c5b9b6845e0a45#diff-6d186b954a58d5bb740f73d84fe39073R89)